lib/std/fs.zig

```diff
@@ -369,6 +370,10 @@ pub const Dir = struct {
             fn nextSolaris(self: *Self) !?Entry {
                 start_over: while (true) {
                     if (self.index >= self.end_index) {
+                        if (self.first) {
+                            std.os.lseek_SET(self.dir.fd, 0) catch unreachable;
```

Comment:
For better debugging, I think adding a comment similar to the ones in the `Iterator.next` functions here would be nice (and the same comment on all the `lseek_SET catch unreachable` calls):

```suggestion
                            std.os.lseek_SET(self.dir.fd, 0) catch unreachable; // EBADF here likely means that the Dir was opened without iteration ability
```

See [this comment](https://github.com/ziglang/zig/issues/10317#issuecomment-1012129796) for a comparison of the stack traces of an uniterable Dir when it happens during `lseek_SET` vs `Iterator.next`

Reply:
Yep, good idea. Will wait for CI results to come through and then push this.